util/exec_test.go

```diff
@@ -91,3 +92,35 @@ func TestFilterArgs(t *testing.T) {
 		},
 	})
 }
+
+func TestParallelExecute(t *testing.T) {
+	type testCase struct {
+		Limit uint
+		Times uint
+
+		Function func()
+	}
+
+	t.Run("Atomic count", func(t *testing.T) {
```

Comment:
So this ensures that the functions are executed correctly. Not sure how to test that the limit works without using `time.Sleep`. I guess you could make something that adds and subtracts from atomic and also checks what the maximum value was. And with a limit of 1 then the maximum value should never exceed 2. But as long as it is functionally correct, we are good I think.